### PR TITLE
Fix bool type comparison

### PR DESCRIPTION
**Pull request by ehsantn**
As title. Fixes some nightly errors.

### CODE REVIEW

**Review by dmsa-or — APPROVED** [review]
Thanks. Scott had suggested maybe writing some unit tests for this function to catch edge cases like this.

File: BodoSQL/bodosql/plan_conversion.py [hunk 1]
```diff
@@ -2476,7 +2476,11 @@ def compare_types(obj_type, expected_type):
         expected_type = pd.api.types.pandas_dtype(expected_type)
 
     if isinstance(obj_type, type) and isinstance(expected_type, type):
-        if issubclass(obj_type, expected_type):
+        if issubclass(obj_type, expected_type) and not (
+            obj_type is bool and expected_type is int
+        ):
+            # bool is a subclass of int in Python, but we don't want to consider them
```
Comment (DrTodd13) [inline BodoSQL/bodosql/plan_conversion.py]: This comment is confusing.  The docstring says return True if subclass or equivalent but here you say we don't want to consider them equivalent yet return true.

**Review by DrTodd13 — APPROVED** [review]

File: BodoSQL/bodosql/plan_conversion.py (hunk 1 above)
Comment (ehsantn) [reply BodoSQL/bodosql/plan_conversion.py]: Moved the comment up to be maybe more clear. We don't consider bool and int equivalent in the Arrow case so has to be consistent and clearly documented. This function needs more work in general after testing these cases.
